addons/project/wizard/project_task_convert_to_subtask.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+class ProjectTaskConvertToSubtask(models.TransientModel):
```

Comment:
I don't think a wizard is needed. Here's a example by your twin where clicking on the option pops up a form :smile: 

https://github.com/odoo/odoo/blob/1a44b849970d3e39111a99ea35db5b586b184dd3/addons/project_todo/views/project_task_views.xml#L152-L184